lib/virtual_host_service_worker/amqp_dispatcher.rb

```diff
@@ -1,32 +1,38 @@
 module VirtualHostServiceWorker
-  
+
   class AmqpDispatcher
-    
+
     ##
     # dispatches amqp-messages to method calls
     #
     def self.dispatch(payload)
       DaemonKit.logger.info("AMQP message received")
-      if payload['ssl_certificate'] and payload['ssl_ca_certificate'] and payload['ssl_key']
-        DaemonKit.logger.info("added a new vhost")
-        
-        if APP_CONFIG['use_haproxy'] == true
-          VirtualHostServiceWorker::HaproxyVHostWriter.setup_v_host(payload)
-        else
-          VirtualHostServiceWorker::NginxVHostWriter.setup_v_host(payload)
-        end
+      begin
+        if payload['ssl_certificate'] and payload['ssl_ca_certificate'] and payload['ssl_key']
+          DaemonKit.logger.info("adding a new vhost: #{payload['server_name']}")
 
-      elsif payload['action'] == 'delete'
-        DaemonKit.logger.info("deleted a vhost")
+          if APP_CONFIG['use_haproxy'] == true
+            VirtualHostServiceWorker::HaproxyVHostWriter.setup_v_host(payload)
+            DaemonKit.logger.info("Succesfully added vhost #{payload['server_name']}")
+          else
+            VirtualHostServiceWorker::NginxVHostWriter.setup_v_host(payload)
+            DaemonKit.logger.info("Succesfully added vhost #{payload['server_name']}")
+          end
 
-        if APP_CONFIG['use_haproxy'] == true
-          VirtualHostServiceWorker::HaproxyVHostWriter.delete_v_host(payload['server_name'])
-        else
-          VirtualHostServiceWorker::NginxVHostWriter.delete_v_host(payload['server_name'])
-        end
+        elsif payload['action'] == 'delete'
+          DaemonKit.logger.info("deleting vhost: #{payload['server_name']}")
 
+          if APP_CONFIG['use_haproxy'] == true
+            VirtualHostServiceWorker::HaproxyVHostWriter.delete_v_host(payload['server_name'])
+            DaemonKit.logger.info("Succesfully deleted vhost: #{payload['server_name']}")
+          else
+            VirtualHostServiceWorker::NginxVHostWriter.delete_v_host(payload['server_name'])
+            DaemonKit.logger.info("Succesfully deleted vhost: #{payload['server_name']}")
+          end
+        end
+      rescue => e
+        DaemonKit.logger.error(e)
```

Comment:
careful with catching **ALL** exceptions here. Its generally fine for logging, but make sure to propagate the error down then. Otherwise this could lead to unexpected behavior.

```ruby
rescue => e
  DaemonKit.logger.error(e)
  # add the raise here
  raise
end
```

For more details: https://stackoverflow.com/questions/10048173/why-is-it-bad-style-to-rescue-exception-e-in-ruby